tensorly/backend/core.py

```diff
@@ -818,22 +800,32 @@ def truncated_svd(self, matrix, n_eigenvecs=None, **kwargs):
             of shape (n_eigenvecs, matrix.shape[1])
             contains the left singular vectors
         """
-        dim_1, dim_2 = matrix.shape
+        dim_1, dim_2 = self.shape(matrix)
         if dim_1 <= dim_2:
             min_dim = dim_1
+            max_dim = dim_2
         else:
             min_dim = dim_2
+            max_dim = dim_1
 
-        if n_eigenvecs is None or n_eigenvecs > min_dim:
-            full_matrices = True
-        else:
-            full_matrices = False
+        if max_dim < n_eigenvecs:
+            warnings.warn(('Trying to compute SVD with n_eigenvecs={0}, which '
+                           'is larger than max(matrix.shape)={1}. Setting '
+                           'n_eigenvecs to {1}').format(n_eigenvecs, max_dim))
+            n_eigenvecs = max_dim
+
+        full_matrices = (n_eigenvecs is None) or (n_eigenvecs > min_dim)
 
-        U, S, V = self.svd(matrix)
+        U, S, V = self.svd(matrix, full_matrices=full_matrices)
         U, S, V = U[:, :n_eigenvecs], S[:n_eigenvecs], V[:n_eigenvecs, :]
         return U, S, V
 
     index = Index()
+
+    @property
```

Comment:
If we go for that solution we could have a function `available_svd_funs` instead of this? What do you think?